src/Microsoft.Health.Fhir.SqlServer/Features/Storage/SqlServerFhirModel.cs

```diff
@@ -106,6 +106,13 @@ public short GetSearchParamId(Uri searchParamUri)
             return _searchParamUriToId[searchParamUri];
         }
 
+        public void SetSearchParamId(string searchParamUri, short searchParamId)
```

Comment:
>SetSearchParamId [](start = 20, length = 16)

It's a bit weird to see how these values been set, but no changes related to how they consumed (or maybe I'm missing them). Was consume part already implemented?

Reply:
Yes, you're right - the `UpsertSearchParams` stored procedure was added in #1430. This change would have been better suited to be part of that PR, but the need to sync up the C# dictionary with the SQL table was only brought to light now that we are relying on the dictionary for reindexing.

All this being said, the dictionary could be updated at different point in the creation of the new search parameter resource (right now `SetSearchParamId()` is being called in `SqlServerSearchParameterStatusDataStore`'s `UpsertStatuses()` method). Very open to feedback on this.

Tagging @feordin here in case you have thoughts/ideas.

Reply:
I think the method makes sense to SetSearchParamId.  I understand @Ivanidzo4ka's comment on it seeming out of place.  Pieces of functionality are coming together over time here.  I do wonder if we need to check for the Uri already existing in the _searchParamUriToId dictionary?  Will the .Add method fail if the key already exists?  As mentioned this method is called from an UpsertStatus method.  Is it possible that could be called more than once for a given SearchParam?

Reply:
Ah yes, good question. I originally had a `.TryAdd()` here but then made modifications to `UpsertSearchParams` in `8.sql` to only return the parameters that were inserted and not updated. Happy to put the `TryAdd()` back in though if it makes more sense to have this logic here.

Reply:
I see the changes in the stored proc.  Just as a question, will there ever be a case when this mapping needs to be updated?  Could the id in Sql ever change for a search parameter uri?

Reply:
I believe this mapping will need to be updated whenever we allow a user to modify/add/delete a `SearchParameter` resource. In the cases of modifying and deleting a search parameter, the flow will be the same as adding one: the request is made by the user, the SQL stored procedure is called, then the C# dictionary is updated to match the search parameter registry in the database.

Reply:
Sounds good.  So perhaps you will have a separate method to update the mapping?  Or will this method be used with an AddOrUpdate?

Reply:
Ah, I see what you're saying - this method is called "SetSearchParamId" but it only adds new ones right now. It might be better suited to be named "AddSearchParamIdMapping" for now, then maybe it could be modified and renamed once update is enabled. Is that correct?

Reply:
Yes, feel free to change it when you add that functionality in the future..

Reply:
Sounds good, renamed to "AddSearchParamIdMapping" in 7c5c5c2, and I plan to modify it once we add the update functionality for SQL in a future PR.